internal/hcs/schema2/launch_options.go

```diff
@@ -0,0 +1,18 @@
+// Autogenerated code; DO NOT EDIT.
+
+// Schema retrieved from branch 'rs_base2_hyp' and build '27645.1002.240617-2119'.
+
+/*
+ * Schema Open API
+ *
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
+ *
+ * API version: 2.4
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+
+package hcsschema
+
+type LaunchOptions struct {
```

Comment:
It doesn't look like this field is actually used anywhere so we should remove it.

Reply:
LaunchOptions definition is required by VirtualMachine structure.
$ go build
# github.com/Microsoft/hcsshim/internal/hcs/schema2
..\..\internal\hcs\schema2\virtual_machine.go:31:19: undefined: LaunchOptions

type VirtualMachine struct {
...
        LaunchOptions   *LaunchOptions   `json:"LaunchOptions,omitempty"`

Reply:
That shouldn't be the case since we didn't have LaunchOptions previously. LaunchOptions is set as omitempty in the VirtualMachine struct, so if it isn't set, the field would just be omitted from the data we send to HCS. Since we aren't setting LaunchOptions anywhere, it should be safe to just remove that field from `VirtualMachine` and remove the unused schema.

Reply:
I agree we should take only the structures that we intend to use. Since the schema is a contract between HCS and containerd, I think we should not modify auto generated code for the schema - either remove parameters, change variable names etc. 
It also becomes challenge for the next person who wants to use and update the schema to know certain variables have to be renamed or omitted.

Reply:
Unfortunately, we cannot avoid needing to modify the schema in some cases. For example, swagger doesn't understand some variable types on golang, so even though the schema in the OS will indicate a certain type, the generated golang code with swagger may output a different type. Additionally, you're also exposing a whole lot of fields that are tagged as private or experimental in the OS that we don't use. The owning team may not want those exposed. 

We should remove any fields we aren't using, and any fields tagged private or experimental that we aren't using.

Reply:
As an example of a type that isn't generated true to the schema, in `VirtualMachineProcessor` the type `Limit` and some other types in that struct are actually nullable in the schema. We don't use them as nullable and so the type being generated as `unit64` is fine, but if we needed it to be nullable, that would require manual intervention.